src/lib/HST/Frontend/Build.hs

```diff
@@ -0,0 +1,33 @@
+module HST.Frontend.Build where
+
+import HST.Frontend.Syntax
+
+name :: String -> Name s
+name = Ident noSrc
+
+var :: Name s -> Exp s l t
+var n = Var noSrc (UnQual noSrc n)
+
+pvar :: Name s -> Pat s l
+pvar = PVar noSrc
+
+caseE :: Exp s l t -> [Alt s l t] -> Exp s l t
+caseE = Case noSrc
+
+alt :: Pat s l -> Exp s l t -> Alt s l t
+alt pat e = Alt noSrc pat (UnGuardedRhs noSrc e) noBinds
+
+binds :: [Decl s l t] -> Binds s l t
+binds = BDecls noSrc
+
+noBinds :: Maybe (Binds s l t)
+noBinds = Nothing
+
+wildcard :: Pat s l
+wildcard = PWildCard noSrc
+
+src :: s -> SrcSpan s
+src = SrcSpan
+
+noSrc :: SrcSpan s
+noSrc = NoSrcSpan
```

Comment:
Okay, but I would like to get a second opinion on this before I remove this module, @ichistmeinname or @stu204146-kiel.

Reply:
First, I like the idea HSE uses that @just95 mentioned in our meeting: make the AST a functor parametrised over the source span information and use `()` if you do not need these information.
Second, as there are plans to use actually use the source span information in the future, the explicit usage of `NoSrcSpan` (or `()`) seems to be more useful in the long run.

Reply:
I think it is better to have smart constructors because they can better tell you what they actually do. And some of them are shorter to write. I would let them.